custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java

```diff
@@ -86,15 +86,18 @@ private static EntryPointConfig parseArgs(String[] args) throws Exception {
     }
 
     public static void main(String[] args) throws Exception {
+        int portNumber = -1;
         try {
             EntryPointConfig config = parseArgs(args);
+            portNumber = config.portNumber;
             PredictorEntryPoint entryPoint = new PredictorEntryPoint(config.className, config.predictorName);
             // TODO: use a specific port, note multiple such gateways might be running.
             GatewayServer gatewayServer = new GatewayServer(entryPoint, config.portNumber);
             gatewayServer.start();
         } catch (Exception e) {
-            System.out.println("Got exception in ComponentEntryPoint: " + e.getMessage());
-            throw e;
```

Comment:
Looks like GatewayServer starts some thread, thus throwing exception doesn't finish the program.
Call exit explicitly.